kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -125,6 +128,33 @@ public SharedInformerFactory withName(String name) {
     return sharedIndexInformerFor(apiTypeClass, inferListType(apiTypeClass), operationContext, resyncPeriodInMillis);
   }
 
+  /**
+   * Constructs and returns a shared index informer with resync period specified for custom resources.
+   *
+   * <b>Note:</b>It watches for events in <b>ALL NAMESPACES</b>.
+   *
+   * @param customResourceContext basic information about the Custom Resource Definition corresponding to that custom resource
+   * @param apiTypeClass apiType class
+   * @param apiListTypeClass api list type class
+   * @param resyncPeriodInMillis resync period in milliseconds
+   * @param <T> the type parameter (should extend {@link io.fabric8.kubernetes.api.model.HasMetadata} and implement {@link io.fabric8.kubernetes.api.model.Namespaced})
+   * @param <L> the type's list parameter (should extend {@link io.fabric8.kubernetes.api.model.KubernetesResourceList}
+   * @return the shared index informer
+   */
+  @Deprecated
```

Comment:
Please add a reference to the options to use instead.

Reply:
That's something I wanted to do, but given the fact that we don't have a clear path, I was unable to elaborate a proper message.

What do you think about "Since 5.x versions of client {@link CustomResourceDefinitionContext} are configured via annotations in CustomResource implementations, please use any of the alternative sharedIndexInformerForCustomResource methods"

Reply:
Sounds good.